test_runner/regress/test_signed_char.py

```diff
@@ -0,0 +1,48 @@
+import os
+
+from fixtures.neon_fixtures import NeonEnv
+
+EXPECTED_OUT_PATH = os.path.join(
+    os.curdir, "test_runner", "regress", "data", "test_signed_char.out"
+)
+SIGNED_CHAR_EXTRACT = """
+    WITH
+  pagenumbers AS (
+    SELECT num FROM generate_series(0, (pg_relation_size('test_payload_idx') / 8192) - 1) it(num)
+  )
+    SELECT num,
+    substr(page, 9, 8192-8),
+    (gin_page_opaque_info(page)).*
+    FROM pagenumbers,
+    LATERAL (SELECT get_raw_page('test_payload_idx', num)) AS p(page)
+    WHERE ARRAY['leaf'] = ((gin_page_opaque_info(page)).flags);
+    """
+
+
+def test_signed_char(neon_simple_env: NeonEnv):
+    env = neon_simple_env
+    endpoint = env.endpoints.create_start("main")
+
+    ses1 = endpoint.connect().cursor()
```

Comment:
I would probably recommend using a Python context block here. Grep for `with` if you're unfamiliar with it.